src/front-end/typescript/lib/pages/proposal/code-with-us/edit/tab/proposal.tsx

```diff
@@ -705,6 +713,47 @@ const update: component_.page.Update<State, InnerMsg, Route> = ({
         ]
       ];
     }
+    case "checkStatus": {
+      const proposal = state.proposal;
+      return [
+        state,
+        proposal && state.isEditing
```

Comment:
I suggest adding a boolean to `state` to indicate if the user has dismissed the `orgProposalChangeDetected` modal.

Once the user has seen and acknowledged the modal, there is no need to show the change detected modal again until they exit form-edit-mode. So, dismissing the modal would set `state.hasSeenChangeDetectedModal = true` (for example).

In this line 720, you could add `&& !state.hasSeenChangeDetectedModal` to the condition to make the polling request to the back-end.

Then, when the user stops editing/saves/cancels, you can reset `state.hasSeenChangeDetectedModal = false` so if the user starts editing again, the polling restarts.

The reason I am making this suggestion is because there may be unbuffered `checkStatus` messages waiting to be evaluated. There is no guarantee that `Cmd`s are executed in order, so handling `Msg`s should try not to make implicit assumptions about which messages will or won't be fired in the future.